torchdata/dataloader2/utils/worker.py

```diff
@@ -84,9 +153,22 @@ def process_reset_fn(
     reset the random state of the ``DataPipe`` graph and the global random states for ``torch``,
     ``random`` and ``numpy``.
     """
+    # Reset non-sharding process first
+    graph = traverse_dps(datapipe)
+    non_sharding_process_dps = find_dps(graph, communication.iter._IterateQueueDataPipes)
```

Comment:
Question: Is it always the case where `_IterateQueueDataPipes` is the only non-sharding process?

Reply:
Yes because we will find the lowest common ancestor of all non-shardable DataPipes in the main process and replace it by this `_IterateQueueDataPipes` in the worker process.
So, it's guaranteed that there is only a single non-sharding process.